examples/data-objects/clicker/src/index.tsx

```diff
@@ -131,19 +134,59 @@ class CounterReactView extends React.Component<CounterProps, CounterState> {
                 <span className="clicker-value-class" id={`clicker-value-${Date.now().toString()}`}>
                     {this.state.value}
                 </span>
-                <button onClick={() => { this.props.counter.increment(1); }}>+</button>
+                <button onClick={() => { this.props.clicker.increment(); }}>+</button>
             </div>
         );
     }
 }
 
 // ----- FACTORY SETUP -----
 
-export const ClickerInstantiationFactory = new DataObjectFactory<Clicker, undefined, undefined, IEvent>(
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const ClickerInstantiationFactory = new DataObjectFactory<Clicker, object, undefined, IClickerEvents>(
     ClickerName,
     Clicker,
     [SharedCounter.getFactory(), TaskManager.getFactory()],
     {},
 );
 
-export const fluidExport = ClickerInstantiationFactory;
+const clickerComponentId = "clicker";
+
+const registryEntries = new Map([
+    ClickerInstantiationFactory.registryEntry,
+]);
+
+const defaultViewRequestHandler: RuntimeRequestHandler =
```

Comment:
This feels like a lot of added complexity to remove IFLuidHTMLView. Is there a plan to simplify this?

Reply:
Good question, two answers:

1. This is partially a "transfer" rather than an "addition".  We're currently hiding the work that a real app author would need to do to wrap the object in container code in the [webpack-fluid-loader](https://github.com/microsoft/FluidFramework/blob/main/packages/tools/webpack-fluid-loader/src/loader.ts#L87).  With this change that wrapping code is not needed, since we supply real container code instead (which is also a more-realistic usage pattern).
2. I've considered adding a new container runtime factory type to directly support a single split model/view component.  However I don't want to formalize the pattern until we can establish a trend first to assess how broadly it would be useful (i.e. real scenarios vs. just demos) and what its requirements might be.  But that might result in the export looking vaguely like

```typescript
export const fluidExport = new ContainerRuntimeFactoryWithDefaultModelAndView(
    factory, // DataObjectFactory,
    bindViewCallback, // callback with DataObject param and returning something that MountableView.canMount()
);
```

Reply:
Is the additional code here then allowing the container runtime factory to still provide a view for the Clicker data object but without it needing to be within the data object class itself?
i.e. even without this section, could the developer still instantiate just a raw Clicker data object and then pass it as a prop to their `ClickerReactView` which they separately import, similar to the dice rollers in external-controller currently?

Reply:
That's right -- this is similar to the `container-views` sample in that regard.  This is only required in order to be supported by `webpack-fluid-loader` which only knows how to deal with `IFluidMountableView` and things that `HTMLViewAdapter.canAdapt()` -- requests must return one of these.

But someone could certainly separately import and do the `external-controller`/`external-views` approach and ignore this container code entirely if they wanted.

Reply:
Makes sense. So this keep its compatible with both models 😄 Thanks for explaining!